### PR TITLE
[MRG] allow setting or disabling warning about too many open views

### LINKED ISSUE #2074: "UserWarning: It seems you have created more than 10 nilearn views." But no way to delete nilearn views
When more than 10 _html_view_ objects via `html_view = nilearn.plotting.view_img()` in a loop, you will see the warning:

> UserWarning: It seems you have created more than 10 nilearn views. As each view uses dozens of megabytes of RAM, you might want to delete some of them.

This is a helpful warning, but it doesn't give the user any way to deal with the problem, as there doesn't seem to be any documented way to free the memory associated with these objects. There is also no clean way (besides e.g. https://stackoverflow.com/questions/5644836/in-python-how-does-one-catch-warnings-as-if-they-were-exceptions/30368735) to disable the warning for users that are sure they have sufficient memory, and/or don't wish to have the console spammed.

It is also not clear to me from the [source code](https://github.com/nilearn/nilearn/blob/2df00ffb1826550b9c8e6497c7c67756e2188039/nilearn/plotting/js_plotting_utils.py#L102) how to work around this: I don't seem to be able to call `html_view._all_open_html_repr.clear()` , `html_view._all_open_html_repr.remove()`, `HTMLDocument._all_open_html_repr.clear()` or etc to manually shrink the size of the `WeakRef` object which has the length causing the error message, although this may be due to my lack of familiarity with these objects. 

Any workaround would be appreciated.

### PR DESCRIPTION
**Pull request by jeromedockes**
closes #2074

### CODE REVIEW

File: nilearn/plotting/js_plotting_utils.py [hunk 1]
```diff
@@ -99,7 +110,12 @@ def __init__(self, html, width=600, height=400):
 
     def _check_n_open(self):
         HTMLDocument._all_open_html_repr.add(self)
-        if len(HTMLDocument._all_open_html_repr) > 9:
+        if MAX_IMG_VIEWS_BEFORE_WARNING is None:
+            return
+        if MAX_IMG_VIEWS_BEFORE_WARNING < 0:
+            return
+        if len(HTMLDocument._all_open_html_repr
```
Comment (kchawla-pi) [inline nilearn/plotting/js_plotting_utils.py]: Can we do:
```python
         if len(HTMLDocument._all_open_html_repr) ... :
                  # display warning
         else:
                  return
```

Comment (jeromedockes) [reply nilearn/plotting/js_plotting_utils.py]: we need to check if `MAX_IMG_VIEWS_BEFORE_WARNING` is not `None` before comparing it to an int. also, this makes it more explicit in which cases the check is not performed

File: nilearn/plotting/js_plotting_utils.py [hunk 2]
```diff
@@ -99,7 +110,12 @@ def __init__(self, html, width=600, height=400):
...
+               ) > MAX_IMG_VIEWS_BEFORE_WARNING - 1:
             warnings.warn('It seems you have created more than 10 '
```
Comment (bthirion) [inline nilearn/plotting/js_plotting_utils.py]: Prbably need to changer this number then.

**Comment by jeromedockes** [comment]
@kchawla-pi any other comments?

**Comment by kchawla-pi** [comment]
Nope.

**Comment by jeromedockes** [comment]
> Nope.

i guess it's ready for merge then

**Comment by kchawla-pi** [comment]
We didn't add a whats_new entry for this . Isn't this a user-facing change?

**Pull request merged**